src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -218,10 +249,12 @@ class HuiMapCard extends LitElement implements LovelaceCard {
       ratio && ratio.w > 0 && ratio.h > 0
         ? `${((100 * ratio.h) / ratio.w).toFixed(2)}%`
         : (root.style.paddingBottom = "100%");
+
+    this._date = new Date();
```

Comment:
We should do this inside `setConfig`. We should also reset the cached history.

Right now we will re-fetch history if config has changed into a config that shows history. This means that you cannot set history to 0.

Reply:
We don't need it anymore. When we change config then `update()` is triggert and `if changedProps.has("_config")` then we fetch the history and while this we update the Date attribute.

Reply:
If new config has `hours_to_show = 0` it won't update the history nor clear it.

Reply:
But `_history` is never cleared.

Reply:
I'm too slow...

Reply:
I've added the history cleanup just now. 

> We don't need it anymore. When we change config then update() is triggert and if  changedProps.has("_config") then we fetch the history and while this we update the Date attribute

This was meant to the line of code above. Sorry for unclear statement